LiteCore/Query/SQLiteFleeceFunctions.cc

```diff
@@ -235,8 +235,17 @@ namespace litecore {
                     {
                         switch ( sqlite3_value_subtype(arg) ) {
                             case 0:
-                                // Untagged blob is already Fleece data
-                                break;
+                                {
+                                    // If the size is 0, it may actually be a Fleece Null with missing subtype.
+                                    if ( sqlite3_value_bytes(arg) == 0 ) {
+                                        slice encoded = Encoder::kPreEncodedNull;
+                                        sqlite3_result_blob(ctx, encoded.buf, int(encoded.size), SQLITE_STATIC);
+                                        return;
+                                    } else {
+                                        // Untagged blob with size is already Fleece data
+                                        break;
+                                    }
+                                }
```

Comment:
Is it equivalent to:
case 0:
case kFleeceNullSubtype:
    if (size > 0) break; // untagged blob ...
   else {
     // Fleece/JSON null
    return;
  }

Reply:
I believe the latest commit is more appropriate